File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/GroupFragAdm/adapterRecVSpecificChapDetailsGrpFAdm.java
package com.example.adminapp.GroupFragAdm;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.recyclerview.widget.RecyclerView;

import com.example.adminapp.R;

import java.util.ArrayList;

public class adapterRecVSpecificChapDetailsGrpFAdm extends RecyclerView.Adapter<adapterRecVSpecificChapDetailsGrpFAdm.myviewholder> {

    ArrayList<dataModelRecVSpecificChapDetailsFAdm> dataHolder;
    ArrayList<dataModelRecVSpecificChapDetailsFAdm> dataHolderBackup;

    public adapterRecVSpecificChapDetailsGrpFAdm(ArrayList<dataModelRecVSpecificChapDetailsFAdm> dataHolder) {
        this.dataHolder = dataHolder;
        dataHolderBackup = new ArrayList<>(dataHolder);
    }

    @NonNull
    @Override
    public myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new myviewholder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.single_row_des_recyclerv_specific_chap_detailsf_adm,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull myviewholder holder, int position) {
        holder.moduleSelectNameTvSpecChapFAdm.setText(dataHolder.get(position).getModuleSelectNameTvSpecChapFAdm());
        holder.moduleDesTvCardLaySpecChapFAdm.setText(dataHolder.get(position).getModuleDesTvCardLaySpecChapFAdm());
        holder.moduleNameInSelectedChapTvCardLaySpecChapFAdm.setText(dataHolder.get(position).getModuleNameInSelectedChapTvCardLaySpecChapFAdm());
    }

    @Override
    public int getItemCount() {
        return dataHolder.size();
    }

    public class myviewholder extends RecyclerView.ViewHolder {
        TextView moduleSelectNameTvSpecChapFAdm,moduleNameInSelectedChapTvCardLaySpecChapFAdm,
                moduleDesTvCardLaySpecChapFAdm,resourcesForThatModuleTvCardLaySpecificChapDetailsFAdm;
        FrameLayout parentFrameLayout;
        RecyclerView parentRecyclerView;

        public myviewholder(@NonNull View itemView) {
            super(itemView);
            moduleSelectNameTvSpecChapFAdm = itemView.findViewById(R.id.moduleSelectNameTvSpecChapFAdm);
            moduleNameInSelectedChapTvCardLaySpecChapFAdm = itemView.findViewById(R.id.moduleNameInSelectedChapTvCardLaySpecChapFAdm);
            moduleDesTvCardLaySpecChapFAdm = itemView.findViewById(R.id.moduleDesTvCardLaySpecChapFAdm);
            resourcesForThatModuleTvCardLaySpecificChapDetailsFAdm = itemView.findViewById(R.id.resourcesForThatModuleTvCardLaySpecificChapDetailsFAdm);

        }
    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/GroupFragAdm/adapterRecVSpecificChapDetailsGrpFAdm.kt
package com.example.adminapp.GroupFragAdm

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.adminapp.R

class adapterRecVSpecificChapDetailsGrpFAdm(var dataHolder: ArrayList<dataModelRecVSpecificChapDetailsFAdm>) :
    RecyclerView.Adapter<adapterRecVSpecificChapDetailsGrpFAdm.myviewholder>() {
    var dataHolderBackup: ArrayList<dataModelRecVSpecificChapDetailsFAdm>

    init {
        dataHolderBackup = ArrayList(dataHolder)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): myviewholder {
        return myviewholder(
            LayoutInflater.from(parent.context)
                .inflate(
                    R.layout.single_row_des_recyclerv_specific_chap_detailsf_adm,
                    parent,
                    false
                )
        )
    }

    override fun onBindViewHolder(holder: myviewholder, position: Int) {
        holder.moduleSelectNameTvSpecChapFAdm.text =
            dataHolder[position].moduleSelectNameTvSpecChapFAdm
        holder.moduleDesTvCardLaySpecChapFAdm.text =
            dataHolder[position].moduleDesTvCardLaySpecChapFAdm
        holder.moduleNameInSelectedChapTvCardLaySpecChapFAdm.text =
            dataHolder[position].moduleNameInSelectedChapTvCardLaySpecChapFAdm
    }

    override fun getItemCount(): Int {
        return dataHolder.size
    }

    inner class myviewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var moduleSelectNameTvSpecChapFAdm: TextView
        var moduleNameInSelectedChapTvCardLaySpecChapFAdm: TextView
        var moduleDesTvCardLaySpecChapFAdm: TextView
        var resourcesForThatModuleTvCardLaySpecificChapDetailsFAdm: TextView
        var parentFrameLayout: FrameLayout? = null
        var parentRecyclerView: RecyclerView? = null

        init {
            moduleSelectNameTvSpecChapFAdm =
                itemView.findViewById(R.id.moduleSelectNameTvSpecChapFAdm)
            moduleNameInSelectedChapTvCardLaySpecChapFAdm =
                itemView.findViewById(R.id.moduleNameInSelectedChapTvCardLaySpecChapFAdm)
            moduleDesTvCardLaySpecChapFAdm =
                itemView.findViewById(R.id.moduleDesTvCardLaySpecChapFAdm)
            resourcesForThatModuleTvCardLaySpecificChapDetailsFAdm =
                itemView.findViewById(R.id.resourcesForThatModuleTvCardLaySpecificChapDetailsFAdm)
        }
    }
}
